axlearn/common/trainer_config_modifier_test.py

```diff
@@ -65,6 +67,26 @@ def test_remat_policy_override(self):
             _ = cfg_modifier(cfg)
 
 
+class ModelConfigModifierTest(test_utils.TestCase):
+    def test_model_config_override(self):
+        cfg = SpmdTrainer.default_config().set(model=causal_lm.Model.default_config())
+        self.assertRegex(str(cfg.model.decoder), ".*StackedTransformerLayer")
+
+        cfg_modifier = (
+            ModelConfigModifier.default_config()
+            .set(
+                model_cfg_modifications={
+                    "model.decoder.transformer": RepeatedTransformerLayer.default_config(),
+                }
+            )
+            .instantiate()
+        )
+
+        cfg = cfg_modifier(cfg)
+        # The default StackedTransformerLayer should have changed to RepeatedTransformerLayer
+        self.assertRegex(str(cfg.model.decoder), ".*RepeatedTransformerLayer")
```

Comment:
Can we do exact match here for the config, something like following: 
```suggestion
        self.assertTrue(cfg.model.decoder is RepeatedTransformerLayer.default_config())
```